cirq/experiments/t2_decay_experiment.py

```diff
@@ -163,21 +200,77 @@ def t2_decay(
 
     y_basis_measurements = results[abs(results.inv_y) > 0]
     x_basis_measurements = results[abs(results.inv_x) > 0]
-    x_basis_tabulation = pd.crosstab(x_basis_measurements.delay_ns,
-                                     x_basis_measurements.output).reset_index()
-    y_basis_tabulation = pd.crosstab(y_basis_measurements.delay_ns,
-                                     y_basis_measurements.output).reset_index()
 
-    # If all measurements are 1 or 0, fill in the missing column with all zeros.
-    for tab in [x_basis_tabulation, y_basis_tabulation]:
-        for col_index, name in [(1, 0), (2, 1)]:
-            if name not in tab:
-                tab.insert(col_index, name, [0] * tab.shape[0])
+    if num_pulses and len(num_pulses) > 1:
+        max_pulses = max(num_pulses)
+        cols = [f'pulse_{t}' for t in range(max_pulses)]
+        x_basis_measurements['num_pulses'] = x_basis_measurements[cols].sum(
+            axis=1)
+        y_basis_measurements['num_pulses'] = y_basis_measurements[cols].sum(
+            axis=1)
+
+    x_basis_tabulation = _create_tabulation(x_basis_measurements)
+    y_basis_tabulation = _create_tabulation(y_basis_measurements)
 
     # Return the results in a container object
     return T2DecayResult(x_basis_tabulation, y_basis_tabulation)
 
 
+def _create_tabulation(measurements):
+    """Returns a sum of 0 and 1 results per index from a list of measurements.
+    """
+    if 'num_pulses' in measurements.columns:
+        cols = [measurements.delay_ns, measurements.num_pulses]
+    else:
+        cols = [measurements.delay_ns]
+    tabulation = pd.crosstab(cols, measurements.output).reset_index()
+    # If all measurements are 1 or 0, fill in the missing column with all zeros.
+    for col_index, name in [(1, 0), (2, 1)]:
+        if name not in tabulation:
+            tabulation.insert(col_index, name, [0] * tabulation.shape[0])
+    return tabulation
+
+
+def _cpmg_circuit(qubit: devices.GridQubit, delay_var: sympy.Symbol,
+                  max_pulses: int):
+    """Creates a CPMG circuit for a given qubit.
+
+    The circuit will look like:
+      sqrt(X) - wait(delay_var)  - X - wait(2*delay_var) - ... - wait(delay_var)
+    with max_pulses number of X gates.
+
+    The X gates are paramterizd by 'pulse_N' symbols so that pulses can be
+    turned on and off.  This is done to combine circuits with different pulses
+    into the same paramterized circuit.
+    """
+    circuit = circuits.Circuit(
+        ops.Y(qubit)**0.5,
+        ops.WaitGate(value.Duration(nanos=delay_var))(qubit), ops.X(qubit))
+    for n in range(max_pulses):
+        pulse_n_on = sympy.Symbol(f'pulse_{n}')
```

Comment:
No action needed: I think I understand why you're doing this. We shouldn't have to. Filed #3266.